File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge40_test.js
const triangle = require("./Challenge40.js");

const tests = {
  input: [1, 2, 3, 4, 0],
  output: [
    [[1]],
    [[1], [1, 1]],
    [[1], [1, 1], [1, 2, 1]],
    [[1], [1, 1], [1, 2, 1], [1, 3, 3, 1]],
    [],
  ],
};

tests.input.forEach((test, index) => {
  console.log("\noriginal:", test);
  const pascalTriangle = triangle.func(test);
  pascalTriangle;
  console.log("expected: ", tests.output[index]);
});

File: app/src/main/java/com/mouredev/weeklychallenge2022/Challenge40_test.rb
# frozen_string_literal: true

require_relative './Challenge40'

TESTS = { input: [1, 2, 3, 4, 0],
          output: [[[1]],
                   [[1], [1, 1]],
                   [[1], [1, 1], [1, 2, 1]],
                   [[1], [1, 1], [1, 2, 1], [1, 3, 3, 1]],
                   'no valid height'] }.freeze

TESTS[:input].each_with_index do |test, index|
  print "\noriginal:", test, "\n"
  Main.new(test).execute
  print "\nexpected:", TESTS[:output][index], "\n"
end

# test private method in class
# pascal_triangle = PascalTriangle.new.send(:cell, 4, 2)
# puts "private pascal_triangle: #{pascal_triangle}"
